mutiny-core/src/federation.rs

```diff
@@ -707,6 +791,25 @@ impl<S: MutinyStorage> FederationClient<S> {
     }
 }
 
+fn maybe_update_after_checking_fedimint<S: MutinyStorage>(
+    updated_invoice: MutinyInvoice,
+    logger: Arc<MutinyLogger>,
+    storage: S,
+) -> Result<(), MutinyError> {
+    Ok(
+        if matches!(
+            updated_invoice.status,
+            HTLCStatus::Succeeded | HTLCStatus::Failed
+        ) {
+            log_debug!(logger, "Saving updated payment");
+            let hash = updated_invoice.payment_hash.into_32();
+            let inbound = updated_invoice.inbound;
+            let payment_info = PaymentInfo::from(updated_invoice);
+            persist_payment_info(&storage, &hash, &payment_info, inbound)?;
+        },
+    )
```

Comment:
this `Ok()` around everything is a weird pattern.

Also doing `if matches!` can just be a `match` statement